core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1049,16 +1045,13 @@ class LogManager(logDirs: Seq[File],
         error(s"Exception in kafka-delete-logs thread.", e)
     } finally {
       try {
-        scheduler.schedule("kafka-delete-logs",
+        scheduler.scheduleOnce("kafka-delete-logs",
           deleteLogs _,
-          delay = nextDelayMs,
-          unit = TimeUnit.MILLISECONDS)
+          nextDelayMs)
       } catch {
         case e: Throwable =>
-          if (scheduler.isStarted) {
```

Comment:
Need to double check if this change makes sense.

Reply:
@junrao What's your thought regarding this? This check seems a bit odd. If the scheduler is not started, we return `NoOpScheduledFutureTask`.

Reply:
Thanks for catching this. #11351 changed the scheduler to return NoOpScheduledFutureTask if the scheduler is not started. But we forgot to change the code here accordingly.